File: core/src/net/natruid/jungle/components/ObstacleComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.PooledWeaver;
import net.natruid.jungle.utils.ObstacleType;

@PooledWeaver
public class ObstacleComponent extends Component {
    public ObstacleType type = ObstacleType.TREE;
    public boolean destroyable = false;
    public float maxHp = 0;
    public float hp = 0;

    protected void reset() {
        type = ObstacleType.TREE;
    }
}

File: core/src/net/natruid/jungle/components/ObstacleComponent.kt
package net.natruid.jungle.components

import com.artemis.Component
import net.natruid.jungle.utils.ObstacleType

class ObstacleComponent : Component() {
    var type: ObstacleType = ObstacleType.TREE
    var destroyable: Boolean = false
    var maxHp: Float = 0f
    var hp: Float = 0f
}
